src/components/body/body-cell.component.ts

```diff
@@ -15,7 +15,7 @@ import { mouseEvent, keyboardEvent } from '../../events';
   template: `
     <div class="datatable-body-cell-label">
       <label
-        *ngIf="column.checkboxable"
+        *ngIf="column.checkboxable && (!displayCheck || displayCheck(row))"
         class="datatable-checkbox">
```

Comment:
@ShinDarth An example from above [comment](https://github.com/swimlane/ngx-datatable/pull/1056/files#r144982935):
change this line to:
`*ngIf="column.checkboxable && (!displayCheck || displayCheck(row, column, value))" `

Reply:
done